repository/src/main/java/com/epam/esm/repository/TagRepository.java

```diff
@@ -0,0 +1,60 @@
+package com.epam.esm.repository;
+
+import com.epam.esm.entity.Tag;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Interface that provides functionality for manipulating {@link Tag} entity.
+ *
+ * @author Geller Egor
+ */
+public interface TagRepository {
+
+    /**
+     * Finds all stored tags
+     *
+     * @return list of {@link Tag} entity.
+     */
+    List<Tag> findAll();

```

Comment:
The CertificateRepository and TagRepository interfaces have common methods, such as findById(), create(). Let's think about abstraction at a higher level